### ISSUE TITLE
Improve README for troubleshooting "project does not exist" 

### ISSUE THREAD

**Issue by eeaton** [terraform-google-modules/terraform-example-foundation#1195]
### TL;DR

Improve troubleshooting guidance in the readme for errors related to a project ID stored in terraform state when the GCP project doesn't exist. This is particularly common when a previous run failed due to project quota issues.

Example of somebody getting stuck, and detailed description of the fix, at: https://github.com/terraform-google-modules/terraform-example-foundation/issues/1187 
https://github.com/terraform-google-modules/terraform-example-foundation/issues/488#issuecomment-973234938

### Terraform Resources

```markdown
n/a
```

### Detailed design

```markdown
n/a
```

### Additional information
See also https://github.com/terraform-google-modules/terraform-example-foundation/issues/451 

Common scenarios where the build can fail during tf apply after passing terraform validate
 - quota issues
 - values for project labels contain illegal character (could also improve this by setting conditional validation at the tf variable to catch it sooner)
 - terraform apply fails due to a timeout
